pootle/core/search/backends/elasticsearch.py

```diff
@@ -26,6 +28,26 @@
 logger = logging.getLogger(__name__)
 
 
+def filter_hits_by_distance(hits, source_text, max_distance=0.3):
+    """Returns ES `hits` filtered according to their Levenshtein distance
+    to the `source_text`.
+
+    Any hits with a normalized distance value (0..1) higher than
+    `max_distance` will be discarded. It's assumed that `hits` is already
+    sorted from higher to lower score.
```

Comment:
You shouldn't assume that. Even more, [score is altered below in this file by using the TM weight](https://github.com/translate/pootle/blob/master/pootle/core/search/backends/elasticsearch.py#L113).